Sources/Shared/Protocols/PathAccessible.swift

```diff
@@ -36,19 +37,50 @@ public extension PathAccessible {
       }
     }
 
-    return result as? JSONDictionary
+    return result as? T
   }
 
-  func path(keyPath: String) -> JSONDictionary? {
-    let kinds: [SubscriptKind] = keyPath.componentsSeparatedByString(".").map {
+  private func resolve<T>(path: String) -> T? {
+    let kinds: [SubscriptKind] = path.componentsSeparatedByString(".").map {
       if let index = Int($0) {
         return .Index(index)
       } else {
         return .Key($0)
       }
     }
 
-    return path(kinds)
+    return resolve(kinds)
+  }
+
+  private func extractKey(path: String) -> (key: String, keyPath: String) {
+    return (key: path.split(".").last!,
+            keyPath: Array(path.split(".").dropLast()).joinWithSeparator("."))
+  }
+
+  func path(path: [SubscriptKind]) -> JSONDictionary? {
+    return resolve(path)
+  }
+
+  func path(keyPath: String) -> JSONDictionary? {
+    return resolve(keyPath)
+  }
+
+  func path(keyPath: String) -> String? {
```

Comment:
Yeah I thought about this as well, right now when working on a demo I use `property` and `path` but both could be done with one method that takes the paths into account and does the appropriate thing for the user instead of having multiple methods. The public API is way to big right now.

`array` and `dictionary` could become hidden methods used by `property`.

I think my vote is on `property` and not `path`. Properties can be resolved with keys or key paths. What do you think?